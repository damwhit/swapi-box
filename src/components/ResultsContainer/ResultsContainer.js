import React from 'react'

function ResultsContainer(props) {
  let results;
  if (props.results.length < 0) {
    results = <article>Pick a category</article>;
  } else {
    results = props.results.map((item) => {
      return <article key={item.name}>{item.name}</article>
    });
    return results
  }
}

export default ResultsContainer;
