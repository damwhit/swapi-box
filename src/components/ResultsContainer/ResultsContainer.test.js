import React from 'react';
import { shallow } from 'enzyme';
import ResultsContainer from './ResultsContainer';

describe('ResultsContainer', () => {
  const mockResources = [
    {name: 'Alderaan', category: 'planets', isFavorite: false},
    {name: 'Dagobah', category: 'planets', isFavorite: false},
    {name: 'Endor', category: 'planets', isFavorite: false},
  ];
  const wrapper = shallow(
    <ResultsContainer 
      category="people"
      onClick={jest.fn()}
      resources={mockResources}
      isLoaded={true}
      error={null}
    />
  );

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
