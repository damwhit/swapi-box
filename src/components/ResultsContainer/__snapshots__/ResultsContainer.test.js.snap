// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsContainer should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsContainer
    category="people"
    error={null}
    isLoaded={true}
    onClick={[MockFunction]}
    resources={
      Array [
        Object {
          "category": "planets",
          "isFavorite": false,
          "name": "Alderaan",
        },
        Object {
          "category": "planets",
          "isFavorite": false,
          "name": "Dagobah",
        },
        Object {
          "category": "planets",
          "isFavorite": false,
          "name": "Endor",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Pick a category",
    },
    "ref": null,
    "rendered": "Pick a category",
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Pick a category",
      },
      "ref": null,
      "rendered": "Pick a category",
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
